stories/module-analytics-setup.stories.js

```diff
@@ -21,22 +21,63 @@
  */
 import { storiesOf } from '@storybook/react';
 
+/**
+ * WordPress dependencies
+ */
+import { createRegistry } from '@wordpress/data';
+
 /**
  * Internal dependencies
  */
+import * as coreForms from '../assets/js/googlesitekit/datastore/forms';
+import * as coreLocation from '../assets/js/googlesitekit/datastore/location';
+import * as coreModules from '../assets/js/googlesitekit/modules';
+import * as coreSite from '../assets/js/googlesitekit/datastore/site';
+import * as coreUi from '../assets/js/googlesitekit/datastore/ui';
+import * as coreUser from '../assets/js/googlesitekit/datastore/user';
+import * as coreWidgets from '../assets/js/googlesitekit/widgets';
+import * as modulesAdSense from '../assets/js/modules/adsense';
+import * as modulesAnalytics from '../assets/js/modules/analytics';
+import * as modulesIdeaHub from '../assets/js/modules/idea-hub';
+import * as modulesOptimize from '../assets/js/modules/optimize';
+import * as modulesPageSpeedInsights from '../assets/js/modules/pagespeed-insights';
+import * as modulesSearchConsole from '../assets/js/modules/search-console';
+import * as modulesTagManager from '../assets/js/modules/tagmanager';
 import ModuleSetup from '../assets/js/components/setup/ModuleSetup';
 import * as fixtures from '../assets/js/modules/analytics/datastore/__fixtures__';
 import { STORE_NAME, ACCOUNT_CREATE, PROFILE_CREATE, PROVISIONING_SCOPE } from '../assets/js/modules/analytics/datastore/constants';
 import { CORE_SITE } from '../assets/js/googlesitekit/datastore/site/constants';
 import { CORE_USER } from '../assets/js/googlesitekit/datastore/user/constants';
 import {
 	WithTestRegistry,
-	createTestRegistry,
 	provideModules,
 	provideModuleRegistrations,
 } from '../tests/js/utils';
 import { generateGTMAnalyticsPropertyStory } from './utils/generate-gtm-analytics-property-story';
 
+const createLegacyTestRegistry = () => {
+	const registry = createRegistry();
+
+	[
+		coreForms,
+		coreLocation,
+		coreModules,
+		coreSite,
+		coreUser,
+		coreUi,
+		coreWidgets,
+		modulesAdSense,
+		modulesAnalytics,
+		modulesIdeaHub,
+		modulesOptimize,
+		modulesPageSpeedInsights,
+		modulesSearchConsole,
+		modulesTagManager,
+	].forEach( ( { registerStore } ) => registerStore?.( registry ) );
+
+	return registry;
+};
+
```

Comment:
I think there is a bug in the `getSetupFlowMode` selector that we can/need to fix as part of this PR. The `getSetupFlowMode` checks whether the `modules/analytics-4` datastore exists to determine whether the `analyics-4` feature is enabled. This is wrong because it should rely on the `isFeatureEnabled( 'ga4setup' )` function call instead. 

Could you please revert changes in this file and fix the `getSetupFlowMode` selector? Just replace the following line with `isFeatureEnabled( 'ga4setup' )` and it should fix the storybook issue: https://github.com/google/site-kit-wp/blob/83a77f51de91b2d6f4ededd0387e63731bb6db5c/assets/js/modules/analytics/datastore/setup-flow.js#L38

Reply:
@eugene-manuilov I implemented that selector and the AC says
`If the modules/analytics-4 store is not available, it should return “legacy”.`
https://github.com/google/site-kit-wp/issues/3170

:smile:  I can see now that checking the feature flag is a better way

I'll change it (and related tests) now